prompb/write/v2/types.proto

```diff
@@ -18,47 +18,84 @@ option go_package = "writev2";
+// TODO(bwplotka): Change URL to Prometheus docs once ready.
 message WriteRequest {
+  // symbols contains de-duplicated array of string elements used for various
+  // items in WriteRequest like labels and some metadata items. To decode
+  // each of those items, referenced, by "ref(s)" suffix, you need to lookup the
+  // actual string by index from symbols array. The order of strings is up to
+  // the client, server should not assume any particular encoding.
   repeated string symbols = 1;
+  // timeseries represents multiple series, potentially from multiple targets.
   repeated TimeSeries timeseries = 2 [(gogoproto.nullable) = false];
 }
 
+// TimeSeries represents multiple series.
 message TimeSeries {
-  // Sorted list of label name-value pair references,
-  // encoded as indices to the strings array.
-  // This list's len is always multiple of 2.
-  repeated uint32 labels_refs = 1 ;
-
+  // labels_refs is a sorted list of label name-value pair references, encoded
+  // as indices to the WriteRequest.symbols array. This list's len is always
+  // a multiple of 2.
+  //
+  // Note that there might be multiple TimeSeries objects in the same
+  // WriteRequests with the same labels e.g. for different exemplars, metadata
+  // or created timestamp.
+  repeated uint32 labels_refs = 1;
+  // samples contain zero or more samples for a given timeseries. For a typical
+  // clients, in healthy cases, there will be only one sample, for ~real
+  // time metric streaming.
+  //
+  // Samples are sorted by time (older first).
+  // TODO(bwplotka): Can/should samples and histograms be in enum? Do we allow mix of those in one TimeSeries?
   repeated Sample samples = 2 [(gogoproto.nullable) = false];
+  // exemplars represents optional set of exemplars attached to this series' samples.
   repeated Exemplar exemplars = 3 [(gogoproto.nullable) = false];
-  // Same as current version
+  // histograms represents zero or more histogram samples for a this series.
+  // TODO(bwplotka): ditto
+  // Histogram samples are sorted by time (older first).
   repeated Histogram histograms = 4 [(gogoproto.nullable) = false];
-
+  // metadata represents metadata associated with the given series' samples.
   Metadata metadata = 5 [(gogoproto.nullable) = false];
-
-  // Optional created timestamp for the metric in ms format,
-  // if the first sample in samples does not contain 0 value.
-  // See model/timestamp/timestamp.go for conversion from time.Time
-  // to Prometheus timestamp.
+  // created_timestamp represents an optional created timestamp associated with
+  // this series' samples in ms format, typically for counter or histogram type
+  // metrics. NOTE: Servers might negotiate "required-created-timestamp" feature
+  // that requires this field to be set for counters and histograms.
+  // For Go, see github.com/prometheus/prometheus/model/timestamp/timestamp.go
+  // for conversion from/to time.Time to Prometheus timestamp.
   int64 created_timestamp = 6;
```

Comment:
We should make this `optional` or take `0` as the null value and make that explicit here. Also, I wonder if we want a `uint64`

Reply:
Nice, I am leaning onto optional field, (similar to https://github.com/prometheus/client_model/blob/master/io/prometheus/client/metrics.proto#L63), but not sure about unit vs int

Reply:
I think int is more fair as we don't want to limit super old time telemetry

Reply:
Looks like we are leaving this as it is (non-optional `int64`). Is it worth adding commentary about timestamp `0` being ignored? Or is that obvious?

Otherwise good to resolve this thread.